replay_build_scripts/upload_build_artifacts.mjs

```diff
@@ -162,15 +162,62 @@ function prepareWindowsBinaries(buildId) {
 function prepareMacOSBinaries(buildId) {
   const buildIdDmgArchive = buildArm ? `${buildId}-arm.dmg` : `${buildId}.dmg`;
   const dmgArchive = buildArm ? "macos-chromium-arm.dmg" : "macos-chromium.dmg";
+  const signedDmg = buildArm
+    ? `${buildId}-arm-signed.dmg`
+    : `${buildId}-signed.dmg`;
   const outdir = buildArm ? "out/Release-ARM" : "out/Release";
   fs.rmSync(path.join(outdir, "Replay-Chromium.app"), {
     recursive: true,
     force: true,
   });
-  fs.renameSync(
-    path.join(outdir, "Chromium.app"),
-    path.join(outdir, "Replay-Chromium.app")
+  const appPath = path.join(outdir, "Replay-Chromium.app");
+  fs.renameSync(path.join(outdir, "Chromium.app"), appPath);
+
+  const originalWorkingDir =
+    process.env["REPLAY_ORIGINAL_WORKING_DIR"] || process.cwd();
+  const codesignPath = process.env["REPLAY_APPLE_CODESIGN_PATH"];
+  const fullCodesignPath = path.join(originalWorkingDir, codesignPath);
+  const p12FilePath = path.join(
+    originalWorkingDir,
+    process.env["REPLAY_APPLE_CODESIGN_CERT_PATH"]
+  );
+  const p12PassPath = path.join(
+    originalWorkingDir,
+    process.env["REPLAY_APPLE_CODESIGN_CERT_PASS_PATH"]
   );
+  const appStoreApiKeyPath = path.join(
+    originalWorkingDir,
+    process.env["REPLAY_APP_STORE_CONNECT_API_KEY_PATH"]
+  );
+  const shouldCodesign = !!codesignPath;
+  if (!shouldCodesign) {
+    console.error("Missing codesigning environment variables", {
+      codesignPath: codesignPath,
+      fullCodesignPath,
+      p12FilePath,
+      p12PassPath,
+      appStoreApiKeyPath,
+    });
+    process.exit(1);
+  }
```

Comment:
the `process.exit(1);` on line 201  gives me pause - there's code below that clearly looks to be able to handle the case where we aren't codesigning (`if (shouldCodesign)` checks below), but the `else`'s will never be reached.  even my suggestion just above wouldn't fix that, since that's still doing the `path.join`'s regardless of if `shouldCodesign` is true or not.  maybe move the path creation (the end of the block in previous comment) down to the `if` branch down on line 251?

```
  if (shouldCodesign) {  // line 251

    ... from above

    const fullCodesignPath = path.join(originalWorkingDir, REPLAY_APPLE_CODESIGN_PATH);
    const p12FilePath = path.join(originalWorkingDir, REPLAY_APPLE_CODESIGN_CERT_PATH);
    ...

    spawnChecked(
      fullCodesignPath,
      ...
...

that would let us get rid of the `process.exit(1)`.

Reply:
I see what you're saying, I like the `|| "missing"` pattern there. I gave a shot at the spirit of what I think you're proposing.